doc/developer/design/20240502_deprecate_subsources.md

```diff
@@ -0,0 +1,190 @@
+FOR ALL TABLES
+```
+
+`postgres_src` is the progress relation; there is no separate
+`postgres_src_progress` relation created. Otherwise this command works the same
+way that it does today.
+
+***
+
+Add a new table to a PostgreSQL source:
+
+```sql
+CREATE TABLE my_new_table FROM SOURCE postgres_src (upstream_schema.upstream_table);
+```
+
+This would replace the `ALTER SOURCE ... ADD {SUBSOURCE|TABLE}` command.
+
+***
+
+Add a new table to a Kafka source:
+
+```sql
+CREATE TABLE kafka_copy FROM SOURCE kafka_src ("t");
+```
+
+At first this looks weird. Why would you ever want two copies of a Kafka source?
+It turns out this lets you incorporate upstream schema changes without losing
+reclocking information! This would get us reusable reclocking
+(https://github.com/MaterializeInc/materialize/issues/17021), without needing to
+introduce a new `SOURCE PROGRESS` concept.
+
+Note that support for this is blocked on at least
+https://github.com/MaterializeInc/materialize/issues/24843, and also additional
+work to support multi-output sources in Kafka.
+
+#### TEXT + IGNORE COLUMNS
+
+These options can remain on `CREATE SOURCE` statements, and the `ADD SUBSOURCE`
+variants of them will move to instead be `CREATE TABLE` variants. We could also
+probably squeeze #26774 in here without much effort.
+
+### Migration
+
+#### Kafka, single-output load generators
+
+- Current source
+    - ID, persist shard becomes read-only `TABLE`
+    - Durable statement becomes `CREATE TABLE curr_source_name FROM SOURCE
+	curr_progress_name ( topic_name )`
+- Progress collection
+    - ID, persist shard becomes `SOURCE`
+    - Durable statement becomes `CREATE SOURCE curr_progress_name FROM...`
+
+#### PG, MySQL, multi-output load generators
+
+- Current source
+    - ID, persist shard becomes regular `TABLE` no longer associated
+	with the source whatsoever. Users can continue depending on this, but they
+	should drop it after the release at their leisure
+    - Durable statement	becomes `CREATE TABLE curr_source_name ()`
+- Subsources
+    - ID, persist shard becomes read-only `TABLE`
+    - Durable statement	becomes `CREATE TABLE curr_source_name FROM SOURCE
+	curr_progress_name ( external_reference )`
+    - `TEXT COLUMNS`, `IGNORE COLUMNS` move from "current source" to `CREATE
+	TABLE`
+- Progress collection
+    - ID, persist hard becomes `SOURCE`
+    - Durable statement becomes `CREATE SOURCE curr_progress_name FROM....`
+
+## Alternatives
+
+The most realistic alternative to this proposal is to leave things as they are.
+
+## Open questions
+
+- Terraform and DBT: Marta, Seth, and Bobby say that we can simply roll forward
+  with a breaking change and teach users what the new version ought to do. Is
+  that right?
+- Do we want to mark the object type for "read-only" tables as something other
+  than "table?" Seems like it would be nice to make clear to someone exploring
+  an environment which tables support inserts vs. which don't.
```

Comment:
I think it depends the interface! I'd suggest we use "table" for anything programmatic—i.e., these tables should be in `mz_tables`, be labeled as objects in `mz_objects`, that sort of thing. But then the commands for humans (`SHOW TABLES`,`SHOW OBJECTS`) probably want to very prominently display read-only vs not.  And the console of course can use a prominent badge or tooltip or whatever makes sense.

Reply:
+1